### ISSUE TITLE
Pyright Will Use non Stubs Package Import if `py.typed` is Present

### ISSUE THREAD

**Issue by mfish33** [microsoft/pyright#11124]
**Describe the bug**
We are trying to use Pyright in a pretty non-standard way, where we can disable certain pyi files via an environment variable. We are implementing this via a conditional `.pth` file.

**Reproduction Setup**
For this example, I am using python 3.9 but any version should work

`/usr/lib/python3.9/site-packages/Foo/__init__.py`:
```
class Bar:
   a = 5
```
`/usr/lib/python3.9/site-packages/Foo/py.typed` (File present)

`/usr/lib/python3.9/site-packages/lsp_redirect.pth`:
```
import sys, os;not os.environ.get("LSP_ONLY_DISABLE") and sys.path.insert(0, "/usr/lib/python3.9/site-packages/lsp")

```

`/usr/lib/python3.9/site-packages/Foo-stubs/__init__.pyi`
```python
class Bar:
   a = 5
   b = 4
```

`example.py`
```python
from Foo import Bar

print( Bar.b )
```

**Output**
```
➜  mypy example.py
Success: no issues found in 1 source file
➜  pyright example.py                                      
/src/TtBase/example.py
  /src/TtBase/example.py:4:12 - error: Cannot access attribute "b" for class "type[Bar]"
    Attribute "b" is unknown (reportAttributeAccessIssue)
```

Currently, you can work around this by adding a py.typed file to `/usr/lib/python3.9/site-packages/Foo-stubs`, but this is confusing, and I only figured this out by reading the ImportResolver source code.

**Comment by mfish33** [comment]
It looks like this issue can be fixed by the following patch. Just trying to help out and cut down the turnaround time 😄 

```
diff --git a/packages/pyright-internal/src/analyzer/importResolver.ts b/packages/pyright-internal/src/analyzer/importResolver.ts
index 14285cfd2..14dececf4 100644
--- a/packages/pyright-internal/src/analyzer/importResolver.ts
+++ b/packages/pyright-internal/src/analyzer/importResolver.ts
@@ -1780,6 +1780,13 @@ export class ImportResolver {
                 return bestImportSoFar;
             }
 
+            // Prefer stub packages of non stub packages. I.E. Foo-stubs over Foo.
+            if (bestImportSoFar.isStubPackage && !newImport.isStubPackage) {
+                return bestImportSoFar;
+            } else if (!bestImportSoFar.isStubPackage && newImport.isStubPackage) {
+                return newImport;
+            }
+
             // Prefer py.typed over non-py.typed.
             if (bestImportSoFar.pyTypedInfo && !newImport.pyTypedInfo) {
                 return bestImportSoFar;
```
